Index: BubbleSort/src/BubbleSort.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/BubbleSort/src/BubbleSort.java b/BubbleSort/src/BubbleSort.java
new file mode 100644
--- /dev/null	(date 1640362015480)
+++ b/BubbleSort/src/BubbleSort.java	(date 1640362015480)
@@ -0,0 +1,31 @@
+import java.util.Arrays;
+
+public class BubbleSort {
+    public static void main(String[] args) {
+    int [] dataArray = {0, 3, 40, 89, -5, 6, 2, 4};
+    for (int [] arr : dataArray) {
+        System.out.print(Arrays.toString(arr) + "=>");
+        BubbleSortAlgorithm.sort(arr);
+        System.out.println(Arrays.toString(arr));
+    }
+
+    }
+}
+
+class BubbleSortAlgorithm {
+    public static void sort(int[] array) {
+        // itN - number of iteration
+        for (int itN = 0; itN < array.length - 1; itN++) {
+            // inner iteration loop
+            // eN - element number
+            for (int eN = array.length - 1;
+                 eN > itN; eN--) {
+                if (array[eN - 1] > array[eN]){
+                    int tmp = array[eN - 1]; // tmp - temporary storage
+                    array[eN - 1] = array[eN];
+                    array[eN] = tmp;
+                }
+            }
+        }
+    }
+ }
Index: BubbleSort/BubbleSort.iml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/BubbleSort/BubbleSort.iml b/BubbleSort/BubbleSort.iml
new file mode 100644
--- /dev/null	(date 1640362015465)
+++ b/BubbleSort/BubbleSort.iml	(date 1640362015465)
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<module type="JAVA_MODULE" version="4">
+  <component name="NewModuleRootManager" inherit-compiler-output="true">
+    <exclude-output />
+    <content url="file://$MODULE_DIR$">
+      <sourceFolder url="file://$MODULE_DIR$/src" isTestSource="false" />
+    </content>
+    <orderEntry type="jdk" jdkName="openjdk-17" jdkType="JavaSDK" />
+    <orderEntry type="sourceFolder" forTests="false" />
+  </component>
+</module>
\ No newline at end of file
